client/gc.go

```diff
@@ -297,16 +297,14 @@ func (a *AllocGarbageCollector) MakeRoomFor(allocations []*structs.Allocation) e
 }
 
 // MarkForCollection starts tracking an allocation for Garbage Collection
-func (a *AllocGarbageCollector) MarkForCollection(ar *AllocRunner) error {
-	if ar == nil {
```

Comment:
It is called in a very limited number of places, all of which handle nil AllocRunners.